### PR TITLE
Wkz/if

### PR DESCRIPTION
**Pull request by wkz**

### CODE REVIEW

**Review by troglobit — APPROVED** [review]
Your part looks great!  I've extended upon it a bit, and in retrospect, that's not what one should do.  GitHub gets confused about who's who.

Maybe you can have a look at it @wkz and we can merge it together in the morning?  Next step after this is to add libnl support to get ifflags, etc. as well as ethtool to fetch counters.

File: qemu/qemu.sh [hunk 1]
```diff
@@ -51,7 +51,7 @@ append_args()
 	echo -n "debug "
     fi
 
-    echo -n "${QEMU_APPEND} ${QEMU_EXTRA_APPEND} "
+    echo -n "${QEMU_APPEND} bonding.max_bonds=0 dummy.numdummies=0 fb_tunnels=none ${QEMU_EXTRA_APPEND} "
```
Comment (wkz) [inline qemu/qemu.sh:54]: As agreed offline: Build these as modules and set these arguments in `/etc/modprobe.d/`. This way, we will get the same behavior in GNS3, on hardware, etc.

**Review by wkz — COMMENTED** [review]
Great stuff!

File: src/confd/src/ietf-interfaces.c [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 const struct srx_module_requirement ietf_if_reqs[] = {
 	{ .dir = YANG_PATH_, .name = "ietf-interfaces", .rev = "2018-02-20" },
-	{ .dir = YANG_PATH_, .name = "iana-if-type", .rev = "2023-01-26" },
```
[the remark this answers is not in the document]
Comment (troglobit) [reply src/confd/src/ietf-interfaces.c]: Added missing yang model and updated .rev in a41343b

File: src/confd/src/srx_val.c [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	len = vsnprintf(NULL, 0, fmt, ap) + 1;
+	va_end(ap);
+
+	xpath = alloca(len);
+	if (!xpath)
+		return NULL;
+
+	va_start(ap, fmt);
+	vsnprintf(xpath, len, fmt, ap);
+	va_end(ap);
```
[the remark this answers is not in the document]
Comment (troglobit) [reply src/confd/src/srx_val.c]: Yes, we should go over all the code and refactor to use `asprintf()` instead.  I'll put up a separate task for that in the backlog, thanks!

File: src/confd/src/srx_val.c [hunk 4]
```diff
@@ -1,37 +1,68 @@
...
+	va_end(apdup);
 
 	xpath = alloca(len);
 	if (!xpath)
-		return NULL;
+		return -1;
+
+	va_copy(apdup, ap);
+	vsnprintf(xpath, len, fmt, apdup);
+	va_end(apdup);
```
[the remark this answers is not in the document]
Comment (troglobit) [reply src/confd/src/srx_val.c]: In backlog on board.

File: src/confd/src/srx_val.c [hunk 5]
```diff
@@ -3,6 +3,43 @@
...
+	va_end(ap);
+
+	val = alloca(len);
+	if (!val)
+		return -1;
+
+	snprintf(xpath, sizeof(xpath), "%s/%s", xpath_base, node);
+	va_start(ap, fmt);
+	vsnprintf(val, len, fmt, ap);
+	va_end(ap);
```
[the remark this answers is not in the document]
Comment (troglobit) [reply src/confd/src/srx_val.c:26]: In backlog on board.

File: qemu/qemu.sh (hunk 1 above)
Comment (troglobit) [reply qemu/qemu.sh]: Fixed in [a3195ef](https://github.com/kernelkit/infix/pull/13/commits/a3195ef6f3396db536ec172214e57f0268a13638)

File: src/confd/src/srx_val.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (troglobit) [reply src/confd/src/srx_val.c:26]: To be handled in a separate PR.

File: src/confd/src/srx_val.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (rical) [reply src/confd/src/srx_val.c]: I'm working on some new auth functionality in this area. I will create some helper functions that uses `vasprintf`, they should work here as well.

**Pull request merged**